mpi-proxy-split/dbg_reproducible/mpi_reproducible.cpp

```diff
@@ -0,0 +1,126 @@
+#include "mpi_reproducible.h"
+
+#include <dlfcn.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <utime.h>
+
+
+/* We use 'static' becuase we don't want the overhead of the compiler
+initializing these to zero each time the function is called. */
+#define MAX_ALLREDUCE_SENDBUF_SIZE (1024 * 1024 * 16) /* 15 MB */
+
+int
+MPI_Init_thread(int *argc, char ***argv, int required, int *provided)
+{
+
```

Comment:
In order to use Fortran constants, you need to call something like `get_fortran_constants()` on initialization method like this one.